### ISSUE TITLE
android:permission attribute inside <intent-filter> tag in AndroidManifest.xml file

### ISSUE THREAD

**Issue by kajay** [IBM-MIL/Xtify-Android-Auto-Example#1]
Android does not support `android:permission` attribute inside `<intent-filter>` tag so use of  `<intent-filter android:permission="com.google.android.c2dm.permission.SEND">` inside AndroidManifest.xml won't protect the receiver component. I have two questions. 1. whats the use of `android:permission` inside `<intent-filter>` if it does not do anything and 2. shouldn't the `android:permission` attribute is declared inside `<receiver>` tag.

**Comment by ecompton3** [comment]
This setup came from the Xtify documentation. I'm not sure why they have it setup this way but it is how they get their service to work. You can see the setup here: [Xtify Docs](http://developer.xtify.com/display/sdk/Getting+Started+with+Google+Cloud+Messaging) and look at step 3c. You could try raising an issue with them to see why they have it setup in this manner

**Comment by kajay** [comment]
yes it will work without any problem because the app is declaring `<uses-permission>` required to receive the protected broadcast intent from the system. The only issue is security. The receiver component is not protected at all and any malicious app can access that component without any difficulty. Thanks for your response and suggestion.
